### PR TITLE
feat(payments-cart): add currency to cart

### PR DESCRIPTION
**Pull request by julianpoy**
## Because

- We want to determine currency based on the tax address provided by the customer

## This pull request

- Automatically determines the tax address and currency based on their IP address
- Allows user to update their tax address, thereby updating their currency

## Issue that this pull request solves

Closes FXA-7584

### CODE REVIEW

File: libs/payments/metrics/src/lib/glean/glean.types.ts [hunk 1]
```diff
@@ -14,11 +14,8 @@ export type CommonMetrics = {
 };
 
 export type CartMetrics = Partial<
-  Pick<ResultCart, 'uid' | 'errorReasonId' | 'couponCode'>
-> & {
-  //TODO - Replace on completion of FXA-7584 and pick from ResultCart
-  currency: string;
-};
+  Pick<ResultCart, 'uid' | 'errorReasonId' | 'couponCode' | 'currency'>
```
Comment (StaberindeZA) [inline libs/payments/metrics/src/lib/glean/glean.types.ts:17]: Sweet! Thank you for making these changes. I'd already forgotten

File: libs/payments/currency/src/lib/currency.config.ts [hunk 2]
```diff
@@ -13,10 +13,18 @@ export class CurrencyConfig {
   )
   @IsObject()
   public readonly taxIds!: { [key: string]: string };
+
+  @Transform(
+    ({ value }) => (value instanceof Object ? value : JSON.parse(value)),
+    { toClassOnly: true }
+  )
+  @IsObject()
+  public readonly currenciesToCountries!: { [key: string]: string[] };
```
Comment (StaberindeZA) [inline libs/payments/currency/src/lib/currency.config.ts:22]: Does this need to be a config variable or could it just be a const map? Changes probably would happen, but won't happen too often. Either way is good.

**Review by StaberindeZA — COMMENTED** [review]
Looks good, just a couple of things missing.
- DB migration script
- Default env vars for the new config.

As a result currently `payments-next` isn't starting up.

File: libs/payments/currency/src/lib/currency.config.ts (hunk 2 above)
Comment (julianpoy) [reply libs/payments/currency/src/lib/currency.config.ts:22]: It's already an env var in auth server, so I figured we'd want the same here. If that's not true, I'm good with using a const.

**Comment by julianpoy** [comment]
> As a result currently payments-next isn't starting up.

Agh, this really should be part of our test suite.

**Comment by StaberindeZA** [comment]
> > As a result currently payments-next isn't starting up.
> 
> Agh, this really should be part of our test suite.

Could update the functional tests to startup `payments-next`.  [this file here](https://github.com/mozilla/fxa/blob/main/packages/functional-tests/scripts/start-services.sh)

Downside is `payments-next` will be started even if no functional tests are run against it.

**Comment by julianpoy** [comment]
I think `nx affected` might do the trick for that. Would only run if affected downstream dependencies change

File: apps/payments/next/.env.development [hunk 3]
```diff
@@ -68,6 +68,10 @@ FIRESTORE_CONFIG__CREDENTIALS__PRIVATE_KEY=
 FIRESTORE_CONFIG__KEY_FILENAME=
 FIRESTORE_CONFIG__PROJECT_ID=
 
+# Currency Config
+CURRENCY_CONFIG__TAX_IDS={}
+CURRENCY_CONFIG__CURRENCIES_TO_COUNTRIES={}
```
Comment (StaberindeZA) [inline apps/payments/next/.env.development:73]: Could we add the currency mapping in here as the default value?

File: libs/payments/ui/src/lib/nestapp/app.module.ts
Comment (StaberindeZA) [inline libs/payments/ui/src/lib/nestapp/app.module.ts:1]: `nestapp/config.ts` needs to be updated as well

**Review by StaberindeZA — COMMENTED** [review]
`payments-next` is still not starting up unfortunately. Most likely due to missing config in `nestapp/config.ts`.

Left 1 more inline comment as well, for setting default values for the env vars in `env.development`

**Review by StaberindeZA — APPROVED** [review]
r+. Thank you for the changes.

Just a reminder to leave a note in the SRE doc that the db migration needs to be run before deploy tomorrow.

**Pull request merged**